.github/workflows/snapshot.yml

```
@@ -42,20 +47,22 @@ jobs:
         run: bin/dfx-stock-test
       - name: Stop dfx
         run: dfx stop
-      - name: Tag if run on a schedule
-        if: ${{ (github.event_name == 'schedule') || (github.ref_name == 'snapshot-schedule') }}
+      - name: Tag if release should be created
+        if: ${{ (github.event_name == 'schedule') || (inputs.make_release)}}
         id: tag
         run: |
           set -euxo pipefail
           tag_name="$(date +release-%Y-%m-%d)"
-          if git show-ref --tags --verify --quiet "refs/tags/${tag_name}";
-          then
-            echo "There is already a tag for today's release: ${tag_name}"
-          else
-            echo "Tagging as '${tag_name}'"
-            git tag "${tag_name}"
-            git push origin "refs/tags/${tag_name}"
-          fi
+          index=0
+          while git show-ref --tags --verify --quiet "refs/tags/${tag_name}"; do
+            echo "Tag ${tag_name} already exists."
+            index=$((index + 1))
+            tag_name="$(date +release-%Y-%m-%d).${index}"
+            echo "Trying tag ${tag_name}."
+          done
```

If possible, comment the purpose/effect of this loop.

```suggestion
          # If there is already a tag with the same name, try appending ".${next_integer}".
          while git show-ref --tags --verify --quiet "refs/tags/${tag_name}"; do
            echo "Tag ${tag_name} already exists."
            index=$((index + 1))
            tag_name="$(date +release-%Y-%m-%d).${index}"
            echo "Trying tag ${tag_name}."
          done
```

PS: This seems orthogonal to the title. Maybe mention this too?

---

Do you mean the PR title?
It's mentioned in the PR description, but it's not orhogonal.
If you only have scheduled releases on different dates, the tag names based on dates never collide.
But if you can create releases any time, you might create multiple on the same day, so we need this collision avoidance.

I've applied your comment suggestion. Thanks.